apps/test/unit/code-studio/components/header/EditableProjectNameTest.js

```diff
@@ -61,6 +61,13 @@ describe('EditableProjectName', () => {
     wrapper.find('.project_save').simulate('click');
     expect(renameSpy.calledOnce).to.be.true;
     expect(renameSpy.calledWith('New Name')).to.be.true;
+
+    // This manual wait-and-update is needed because a rename is an async operation,
+    // which we're faking in our test, and enzyme doesn't always re-render when needed
+    // in this situation.
+    await new Promise(resolve => setTimeout(resolve, 0));
+    wrapper.update();
```

Comment:
Before, our fake `dashboard.project.rename` method, above, invoked its callback synchronously; that is, the callback happened immediately on the same tick.  Now that it returns a Promise, the combination of `then()` call and nested `setState` call wasn't happening synchronously in the test.  So now we wait one tick for the appropriate `setState` to complete, and (because newer versions of Enzyme don't always re-render like you'd expect it to in these situations) manually asking for another render as well.